lib/Conversion/ImportVerilog/Structure.cpp

```diff
@@ -130,6 +130,7 @@ struct MemberVisitor {
       // connection for the port.
       if (!expr) {
         auto *port = con->port.as_if<PortSymbol>();
+        port = context.updatePortsSymbol(moduleLowering, port);
```

Comment:
Instead of using `updatePortsSymbol` which linearly scans through the ports, could you create a `DenseMap` lookup table right after getting `moduleLowering` which associates the ports with their `SyntaxNode *`? Something like:
```
DenseMap<const slang::syntax::SyntaxNode *, const Symbol *> portsBySyntaxNode;
for (auto port : moduleLowering->ports)
  portsBySyntaxNode.insert({port.getSyntax(), port});
```
We could also store that in the lowering directly.

Then instead of calling `updatePortsSymbol` multiple times, you can map from the instance's port to the module lowering's port once at the beginning of the for loop after `expr = ...`:
```
const auto *expr = con->getExpression();
const auto *port = portsBySyntaxNode.lookup(con->port.getSyntax());
assert(port);
```
After that the original code should just work without requiring more port remapping.

In case the ports don't have a proper `SyntaxNode` associated with them, you can also use the port's name as a key for the lookup table. But the syntax node would be cleaner and faster 🙂

Reply:
Cool! That will avoid duplicate mapping.